File: src/main/java/com/linebot/model/qiita/QiitaRequestParameter.java
package com.linebot.model.qiita;

import com.linebot.model.common.RequestParameter;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.util.MultiValueMap;

@Component
@Data
@Builder
@NoArgsConstructor
public class QiitaRequestParameter implements RequestParameter {
    public QiitaRequestParameter(Integer page, Integer perPage) {
        this.page = page;
        this.perPage = perPage;
    }

    private Integer page;
    private Integer perPage;

    @Override
    public MultiValueMap<String, String> convert() {
        MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
        add(params, "page", this.page);
        add(params, "per_page", this.perPage);
        return params;
    }
}

File: src/main/kotlin/com/linebot/model/qiita/QiitaRequestParameter.kt
package com.linebot.model.qiita

import com.linebot.model.common.RequestParameter
import org.springframework.stereotype.Component
import org.springframework.util.LinkedMultiValueMap
import org.springframework.util.MultiValueMap

@Component
data class QiitaRequestParameter(
    var page: Int?,
    var perPage: Int?
) : RequestParameter {
    override fun convert(): MultiValueMap<String, String> {
        val params = LinkedMultiValueMap<String, String>()
        add(params, "page", this.page)
        add(params, "per_page", this.perPage)
        return params
    }
}
